docs/Training-on-Amazon-Web-Service.md

```diff
@@ -1,40 +1,144 @@
-* [Installing TensorFlow on an AWS EC2 P2 GPU Instance](http://expressionflow.com/2016/10/09/installing-tensorflow-on-an-aws-ec2-p2-gpu-instance/)   
-* [Updating Nvidia CUDA to 8.0.x in Ubuntu 16.04 – EC2 Gx instance](https://aichamp.wordpress.com/2016/11/09/updating-nvidia-cuda-to-8-0-x-in-ubuntu-16-04-ec2-gx-instance/)
+Current limitations of the Unity Engine require that a screen be available to render to when using visual observations. In order to make this possible when training on a remote server, a virtual screen is required. We can do this by installing Xorg and creating a virtual screen. Once installed and created, we can display the Unity environment in the virtual environment, and train as we would on a local machine. Ensure that `headless` mode is disabled when building linux executables which use visual observations.
 
-## Installing ML-Agents
+1. Install and setup Xorg:
 
-1. Move `python` sub-folder of this ml-agents repo to the remote ECS instance, and set it as the working directory.
-2. Install the required packages with `pip3 install .`.
+    ```
+    //Install Xorg
+    sudo apt-get update
+    sudo apt-get install -y xserver-xorg mesa-utils
+    sudo nvidia-xconfig -a --use-display-device=None --virtual=1280x1024
+    
+    //Get the BusID information
+    nvidia-xconfig --query-gpu-info
+    
+    //Add the BusID information to your /etc/X11/xorg.conf file 
+    sudo sed -i 's/    BoardName      "Tesla K80"/    BoardName      "Tesla K80"\n    BusID          "0:30:0"/g' /etc/X11/xorg.conf
+    
+    //Remove the Section "Files" from the /etc/X11/xorg.conf file
+    sudo vim /etc/X11/xorg.conf //And remove two lines that contain Section "Files" and EndSection. 
+    ```
 
-## Testing
+2. Update and setup Nvidia driver:
 
-To verify that all steps worked correctly:
+    ```
+    //Download and install the latest Nvidia driver for ubuntu
+    wget http://download.nvidia.com/XFree86/Linux-x86_64/390.67/NVIDIA-Linux-x86_64-390.67.run
+    sudo /bin/bash ./NVIDIA-Linux-x86_64-390.67.run --accept-license --no-questions --ui=none
+    
+    //Disable Nouveau as it will clash with the Nvidia driver
+    sudo echo 'blacklist nouveau'  | sudo tee -a /etc/modprobe.d/blacklist.conf
+    sudo echo 'options nouveau modeset=0'  | sudo tee -a /etc/modprobe.d/blacklist.conf
+    sudo echo options nouveau modeset=0 | sudo tee -a /etc/modprobe.d/nouveau-kms.conf
+    sudo update-initramfs -u
+    ```
 
+2. Restart the EC2 instance:
+
+    ```
+    sudo reboot now
+    ```
+
+3. Make sure there are no Xorg processes running:
+
+   ```
+   //Kill any possible running Xorg processes
+   //Note that you might have to run this command multiple times depending on how Xorg is configured.  
+   sudo killall Xorg
+   
+   //Check if there is any Xorg process left
+   //You will have a list of processes running on the GPU, Xorg should not be in the list.
+   nvidia-smi
```

Comment:
__Maybe__ show what it should look like when `Xorg` is not in the list ?

Reply:
Maybe I will put two screen shots here?